File: typescript/examples/modify_order.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Modify Order Example
 *
 * Place a resting order and then modify its price.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Place a resting order
  const mid = await sdk.getMid("BTC");
  const limitPrice = Math.floor(mid * 0.97);
  const order = await sdk.buy("BTC", { notional: 11, price: limitPrice, tif: "gtc" });
  console.log(`Placed order at $${limitPrice.toLocaleString()}`);
  console.log(`  OID: ${order.oid}`);

  // Modify to a new price (4% below mid)
  const newPrice = Math.floor(mid * 0.96);
  const newOrder = await order.modify({ price: newPrice });
  console.log(`Modified to $${newPrice.toLocaleString()}`);
  console.log(`  New OID: ${newOrder.oid}`);

  // Clean up
  await newOrder.cancel();
  console.log("Order cancelled.");
}

main().catch(console.error);

File: python/examples/modify_order.py
#!/usr/bin/env python3
"""
Modify Order Example

Place a resting order and then modify its price.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python modify_order.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Modify Order Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Get current price
    mid = sdk.get_mid("BTC")
    print(f"BTC mid price: ${mid:,.2f}")

    # Place a resting order
    limit_price = int(mid * 0.97)
    order = sdk.buy("BTC", notional=11, price=limit_price, tif="gtc")
    print(f"Placed order at ${limit_price:,}")
    print(f"  OID: {order.oid}")

    # Modify to a new price (4% below mid)
    new_price = int(mid * 0.96)
    new_order = order.modify(price=new_price)
    print(f"Modified to ${new_price:,}")
    print(f"  New OID: {new_order.oid}")

    # Clean up
    new_order.cancel()
    print("Order cancelled.")


if __name__ == "__main__":
    main()
